server/api/models.py

```diff
@@ -51,7 +55,78 @@ class Spell(models.Model):
     archetype = models.TextField()
     circles = models.TextField()
 
+class Archetype(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1)
+    name = models.TextField(primary_key=True)
+    desc = models.TextField()
+
+class CharClass(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1, primary_key=True)
+    name = models.TextField()
+    hit_dice = models.TextField()
+    hp_at_1st_level = models.TextField()
+    hp_at_higher_levels = models.TextField()
+    prof_armor = models.TextField()
+    prof_weapons = models.TextField()
+    prof_tools = models.TextField()
+    prof_saving_throws = models.TextField()
+    prof_skills = models.TextField()
+    equipment = models.TextField()
+    table = models.TextField()
+    desc = models.TextField()
+    spellcasting_ability = models.TextField()
+    subtypes_name = models.TextField()
+    archetype = models.ForeignKey(Archetype, on_delete=models.CASCADE, null=True)
+
+class SubRace(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1, primary_key=True)
+    name = models.TextField()
+    desc = models.TextField()
+    asi_desc = models.TextField()
+    asi = JSONField()
+    traits = models.TextField()
+
+class Race(models.Model):
+    slug = models.SlugField(unique=True, default=uuid.uuid1, primary_key=True)
+    name = models.TextField()
+    desc = models.TextField()
+    asi_desc = models.TextField()
+    asi = JSONField()
+    age = models.TextField()
+    alignment = models.TextField()
+    size = models.TextField()
+    speed = JSONField()
+    speed_desc = models.TextField()
+    languages = models.TextField()
+    vision = models.TextField()
+    traits = models.TextField()
+    subtypes  = models.TextField()
+    sub_race = models.ForeignKey(SubRace, on_delete=models.CASCADE, null=True)
```

Comment:
Similar to the Archetype vs Character Class discussion, this foreign key should be on the sub-race. i.e. when you have a tree and you can only store one pointer on each node, you wanna use that pointer to define the parent :)